File: bike java/Bike.java
public class Bike {
    private int accelerator;
    private int gearStatus;
    private boolean running;

    public Bike() {
        this.accelerator = 0;
        this.gearStatus = 0;
        this.running = false;
    }

    public int turnOn() {
        this.running = true;
        return this.gearStatus;
    }

    public int turnOff() {
        this.running = false;
        this.accelerator = 0;
        this.gearStatus = 0;
        return this.gearStatus;
    }

    public Integer accelerate() {
        if (!running) return null;
        if (gearStatus == 0) gearStatus = 1;
        accelerator += gearStatus;
        updateGear();
        return accelerator;
    }

    public Integer decelerate() {
        if (!running) return null;
        accelerator -= gearStatus;
        if (accelerator < 0) accelerator = 0;
        updateGear();
        return accelerator;
    }

    private void updateGear() {
        if (!running) {
            gearStatus = 0;
            return;
        }
        if (accelerator == 0) {
            gearStatus = 0;
        } else if (accelerator >= 41) {
            gearStatus = 4;
        } else if (accelerator >= 31) {
            gearStatus = 3;
        } else if (accelerator >= 21) {
            gearStatus = 2;
        } else {
            gearStatus = 1;
        }
    }


    public int getGearStatus() {
        return gearStatus;
    }


    public boolean isRunning() {
        return running;
    }
}

File: bike javascript/bike.js
function createBike(accelerator = 0, gearStatus = 0) {
  return {
    accelerator: accelerator, 
    gearStatus: gearStatus, 
    deceleration: 0, 
    running: false, 
  };
}

function turnOn(bike) {
  bike.running = true;
  return bike.gearStatus;
}

function turnOff(bike) {
  bike.running = false;
  bike.accelerator = 0;
  bike.gearStatus = 0;
  return bike.gearStatus;
}

function accelerate(bike) {
  if (!bike.running) return null;
  if (bike.gearStatus === 0) bike.gearStatus = 1;
  bike.accelerator += bike.gearStatus;
  updateGear(bike);
  return bike.accelerator;
}

function decelerate(bike) {
  if (!bike.running) return null;
  bike.accelerator -= bike.gearStatus;
  if (bike.accelerator < 0) bike.accelerator = 0;
  updateGear(bike);
  return bike.accelerator;
}

function updateGear(bike) {
  if (!bike.running) {
    bike.gearStatus = 0;
    return;
  }
  if (bike.accelerator === 0) {
    bike.gearStatus = 0;
  } else if (bike.accelerator >= 41) {
    bike.gearStatus = 4;
  } else if (bike.accelerator >= 31) {
    bike.gearStatus = 3;
  } else if (bike.accelerator >= 21) {
    bike.gearStatus = 2;
  } else {
    bike.gearStatus = 1;
  }
}

module.exports = {createBike,turnOn,turnOff,accelerate,decelerate,updateGear,
};
